core/src/main/java/hudson/cli/UnprotectedCLICommand.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2018, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package hudson.cli;
+
+/**
+ * Marker interface used by {@link hudson.cli.CLICommand} implementations to indicate that
```

Comment:
This could also be an annotation. Both require reflection, right?

Reply:
There is no reflection involved here, I don't understand your point :) I am just checking `instanceOf UnprotectedCLICommand` instead of putting direct implementation in the test. That allow also the plugin to implement it themselves.

Reply:
From an implementation point of view, checking the runtime type of an object is reflection. I'm not sure if an `instanceof` check is faster than a `isAnnotationPresent`, but they both accomplish similar goals when making a marker on a class.

Reply:
@jvz yeah ok I see what you mean. I thought you were talking about "pure reflection". I am more fan of interface in the case we need to add some methods to them, it's a lots easier than with annotations.